pkg/glusterutils/peers_gd1.go

```diff
@@ -0,0 +1,79 @@
+package glusterutils
+
+import (
+	"encoding/xml"
+	"os/exec"
+)
+
+type peerGlusterd1 struct {
+	PeerID    string   `xml:"uuid"`
+	Hostname  []string `xml:"hostname"`
+	Connected int      `xml:"connected"`
+	State     int      `xml:"state"`
+	StateStr  string   `xml:"stateStr"`
+}
+
+type peersGlusterd1 struct {
+	XMLName xml.Name        `xml:"cliOutput"`
+	List    []peerGlusterd1 `xml:"peerStatus>peer"`
+}
+
+// PeersGD1 returns the list of peers ( for GlusterD1 )
+func PeersGD1(config *Config) ([]Peer, error) {
+	var gd1Peers peersGlusterd1
+	var peersgd1 []Peer
+	out, err := exec.Command(config.GlusterCmd, "pool", "list", "--xml").Output()
+	if err != nil {
+		return peersgd1, err
+	}
+	err = xml.Unmarshal(out, &gd1Peers)
+	if err != nil {
+		return peersgd1, err
+	}
+	peersgd1 = make([]Peer, len(gd1Peers.List))
+	var online bool
+	// Convert to required format
+	for pidx, peergd1 := range gd1Peers.List {
+		if peergd1.Connected == 1 {
+			online = true
+		} else {
+			online = false
+		}
+		peersgd1[pidx] = Peer{
+			ID:            peergd1.PeerID,
+			PeerAddresses: peergd1.Hostname,
+			Online:        online,
+		}
+	}
+
+	return peersgd1, nil
+}
+
+// IsleaderGD1 returns true or false based on whether the node is the leader of the cluster or not ( for GlusterD1 )
+func IsleaderGD1(config *Config) (bool, error) {
```

Comment:
two separate functions `IsleaderGD1` and `IsleaderGd2` not required. You can combine into one in the `exported.go` file. Use the same approach to decide leader.

Reply:
you can move this function itself as `IsLeader` in exported file. instead of `PeersGD1` call `Peers(config)`